synapse/handlers/register.py

```diff
@@ -220,8 +220,26 @@ def register(
 
         # auto-join the user to any rooms we're supposed to dump them into
         fake_requester = create_requester(user_id)
+
+        # try to create the room if we're the first user on the server
+        if self.hs.config.autocreate_auto_join_rooms:
+            count = yield self.store.count_all_users()
+            auto_create_rooms = count == 1
+
         for r in self.hs.config.auto_join_rooms:
             try:
+                if auto_create_rooms and RoomAlias.is_valid(r):
+                    room_creation_handler = self.hs.get_room_creation_handler()
```

Comment:
can you put this in the constructor?

Reply:
The first part needs to be on a request basis - I guess the self.hs.get_room_creation_handler() could but this is the only place it is used.

Reply:
I know it's the only place it's used, but we prefer to get the handlers in the constructors where possible, so that we detect problems earlier.